### DISCUSSION TITLE
How do I validate rest parameter using function schema?

### DISCUSSION THREAD

**Discussion by d-mok** [colinhacks/zod#1698] [Q&A]
How do I validate a function like:
```typescript
function sum(...numbers: number[]): number {
  return numbers.reduce((a,b)=>a+b)
}
```
using function schema?

Here is a failed attempt:
```typescript
const sum = z
  .function()
  .args( ... z.number() ) 
```

**Comment by JacobWeisenburger** [comment]
Thanks for asking your question the way you did. Not a lot of people do. This is the perfect way to get an answer:
1. Say what you are trying to do
2. Show what you have tried

Perfection! :D

---
But unfortunately, we don't have a solid answer for this yet. We are tracking this issue here: [variadic args function #1282](https://github.com/colinhacks/zod/issues/1282)
